modules/images/webp-uploads/load.php

```diff
@@ -42,60 +46,59 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
-		// This image size does not exist on the defined sizes.
-		if ( ! isset( $image_sizes[ $size_name ] ) || ! is_array( $image_sizes[ $size_name ] ) ) {
+	$dirname = pathinfo( $file, PATHINFO_DIRNAME );
+	foreach ( $metadata['sizes'] as $size_name => $properties ) {
+		// This image size is not defined or not an array.
+		if ( ! is_array( $properties ) ) {
 			continue;
 		}
 
-		$current_size = $image_sizes[ $size_name ];
-		$sources      = array();
-		if ( isset( $current_size['sources'] ) && is_array( $current_size['sources'] ) ) {
-			$sources = $current_size['sources'];
-		}
-
 		// Try to find the mime type of the image size.
 		$current_mime = '';
-		if ( isset( $current_size['mime-type'] ) ) {
-			$current_mime = $current_size['mime-type'];
-		} elseif ( isset( $current_size['file'] ) ) {
-			$current_mime = wp_check_filetype( $current_size['file'] )['type'];
+		if ( isset( $properties['mime-type'] ) ) {
+			$current_mime = $properties['mime-type'];
+		} elseif ( isset( $properties['file'] ) ) {
+			$current_mime = wp_check_filetype( $properties['file'] )['type'];
 		}
 
+		// The mime type can't be determined.
 		if ( empty( $current_mime ) ) {
 			continue;
 		}
 
-		$sources[ $current_mime ] = array(
-			'file'     => isset( $current_size['file'] ) ? $current_size['file'] : '',
-			'filesize' => 0,
-		);
+		// Ensure a `sources` property exists on the existing size.
+		if ( empty( $properties['sources'] ) || ! is_array( $properties['sources'] ) ) {
+			$properties['sources'] = array();
+		}
 
-		// Set the filesize from the current mime image.
-		$file_location = path_join( $dirname, $sources[ $current_mime ]['file'] );
-		if ( file_exists( $file_location ) ) {
-			$sources[ $current_mime ]['filesize'] = filesize( $file_location );
+		if ( empty( $properties['sources'][ $current_mime ] ) ) {
+			$properties['sources'][ $current_mime ] = array(
+				'file'     => isset( $properties['file'] ) ? $properties['file'] : '',
+				'filesize' => 0,
+			);
+			// Set the filesize from the current mime image.
+			$file_location = path_join( $dirname, $properties['file'] );
+			if ( file_exists( $file_location ) ) {
+				$properties['sources'][ $current_mime ]['filesize'] = filesize( $file_location );
+			}
+			$metadata['sizes'][ $size_name ] = $properties;
+			wp_update_attachment_metadata( $attachment_id, $metadata );
 		}
 
 		$formats = isset( $valid_mime_transforms[ $current_mime ] ) ? $valid_mime_transforms[ $current_mime ] : array();
 
 		foreach ( $formats as $mime ) {
-			if ( empty( $sources[ $mime ] ) ) {
+			if ( empty( $properties['sources'][ $mime ] ) ) {
 				$source = webp_uploads_generate_image_size( $attachment_id, $size_name, $mime );
 				if ( is_array( $source ) ) {
-					$sources[ $mime ] = $source;
+					$properties['sources'][ $mime ]  = $source;
+					$metadata['sizes'][ $size_name ] = $properties;
+					wp_update_attachment_metadata( $attachment_id, $metadata );
 				}
 			}
```

Comment:
Nit-pick, but code tends to be easier to read when avoiding excessive nesting. For readability I'd suggest changing this around, so that the check in the beginning is the opposite:
```php
if ( ! empty( $properties['sources'][ $mime ] ) ) {
	continue;
}
```

Then the other code below doesn't need to be nested/indented. The same applies to the `is_array` check, which I furthermore think we should change to use `is_wp_error`, which is more clear. We don't need to check whether the result is an array then, because we know it will always be either a `WP_Error` or an array. So here I'd suggest to do the following:
```php
if ( is_wp_error( $source ) ) {
	continue;
}
```

This way we have even less nesting/indentation.